apps/ejabberd/test/mongooseim_metrics_SUITE.erl

```diff
@@ -0,0 +1,90 @@
+-module(mongooseim_metrics_SUITE).
+
+-include_lib("exml/include/exml.hrl").
+-include_lib("proper/include/proper.hrl").
+-include_lib("eunit/include/eunit.hrl").
+-include_lib("ejabberd/include/jlib.hrl").
+-include_lib("common_test/include/ct.hrl").
+
+-compile([export_all]).
+
+all() -> [subscriptions_initialised].
+
+init_per_suite(C) ->
+    application:load(exometer),
+    application:set_env(exometer, mongooseim_report_interval, 1000),
+    {Port, Socket} = carbon_server:start(1, 0),
+    Sup = spawn(fun() ->
+                        mim_ct_sup:start_link(ejabberd_sup),
+                        Hooks =
+                        {ejabberd_hooks,
+                         {ejabberd_hooks, start_link, []},
+                         permanent,
+                         brutal_kill,
+                         worker,
+                         [ejabberd_hooks]},
+                        C2SSupervisor =
+                        {ejabberd_c2s_sup,
+                         {ejabberd_tmp_sup, start_link, [ejabberd_c2s_sup, ejabberd_c2s]},
+                         permanent,
+                         infinity,
+                         supervisor,
+                         [ejabberd_tmp_sup]},
+                        supervisor:start_child(ejabberd_sup, Hooks),
+                        supervisor:start_child(ejabberd_sup, C2SSupervisor),
+                        receive
+                            stop ->
+                                ok
+                        end
+                end),
+    Reporters = get_reporters_cfg(Port),
+    application:set_env(exometer, report, Reporters),
+    PortServer = receive
```

Comment:
I think this receive could be encapsulated inside the `carbon_sever` module- ex. wait_wait_for_client or sth similar. Btw we can name it carbon_cache_server. Initially, I thought that it might have something to do with carbon copies.